src/host/outputStream.cpp

```diff
@@ -383,221 +376,336 @@ bool ConhostInternalGetSet::PrivateSetScrollingRegion(const SMALL_RECT& scrollMa
-bool ConhostInternalGetSet::GetConsoleOutputCP(unsigned int& codepage)
+// - the outputCP of the console.
+unsigned int ConhostInternalGetSet::GetConsoleOutputCP() const
 {
-    DoSrvGetConsoleOutputCodePage(codepage);
-    return true;
+    return ServiceLocator::LocateGlobals().getConsoleInformation().OutputCP;
 }
 
 // Routine Description:
-// - Connects the PrivateSuppressResizeRepaint API call directly into our Driver
-//      Message servicing call inside Conhost.exe
+// - Forces the VT Renderer to NOT paint the next resize event. This is used by
+//      InteractDispatch, to prevent resizes from echoing between terminal and host.
 // Arguments:
 // - <none>
 // Return Value:
-// - true if successful (see DoSrvPrivateSuppressResizeRepaint). false otherwise.
-bool ConhostInternalGetSet::PrivateSuppressResizeRepaint()
+// - <none>
+void ConhostInternalGetSet::SuppressResizeRepaint()
 {
-    return SUCCEEDED(DoSrvPrivateSuppressResizeRepaint());
+    auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    THROW_IF_FAILED(gci.GetVtIo()->SuppressResizeRepaint());
 }
 
 // Routine Description:
-// - Connects the IsConsolePty call directly into our Driver Message servicing call inside Conhost.exe
-// - NOTE: This ONE method behaves differently! The rest of the methods on this
-//   interface return true if successful. This one just returns the result.
+// - Checks if the console host is acting as a pty.
 // Arguments:
-// - isPty: receives the bool indicating whether or not we're in pty mode.
+// - <none>
 // Return Value:
 // - true if we're in pty mode.
 bool ConhostInternalGetSet::IsConsolePty() const
 {
-    bool isPty = false;
-    DoSrvIsConsolePty(isPty);
-    return isPty;
+    return ServiceLocator::LocateGlobals().getConsoleInformation().IsInVtIoMode();
 }
 
-bool ConhostInternalGetSet::DeleteLines(const size_t count)
+// Routine Description:
+// - Deletes lines in the active screen buffer.
+// Parameters:
+// - count - the number of lines to delete
+void ConhostInternalGetSet::DeleteLines(const size_t count)
 {
-    DoSrvPrivateDeleteLines(count);
-    return true;
+    _modifyLines(count, false);
 }
 
-bool ConhostInternalGetSet::InsertLines(const size_t count)
+// Routine Description:
+// - Inserts lines in the active screen buffer.
+// Parameters:
+// - count - the number of lines to insert
+void ConhostInternalGetSet::InsertLines(const size_t count)
 {
-    DoSrvPrivateInsertLines(count);
-    return true;
+    _modifyLines(count, true);
+}
+
+// Routine Description:
+// - internal logic for adding or removing lines in the active screen buffer
+//   this also moves the cursor to the left margin, which is expected behavior for IL and DL
+// Parameters:
+// - count - the number of lines to modify
+// - insert - true if inserting lines, false if deleting lines
+void ConhostInternalGetSet::_modifyLines(const size_t count, const bool insert)
+{
+    auto& screenInfo = _io.GetActiveOutputBuffer();

```

Comment:
The `GetActiveBuffer` thing wasn't the bug - those calls were all just redundant, because `GetActiveOutputBuffer` is already required to return the active buffer (in some cases it's literally an alias for `GetActiveBuffer`).

The bug was that the original code was getting the buffer from the global `CONSOLE_INFORMATION` class, which is not necessarily the same buffer as the one obtained from `_io` (which is tracking the current output handle).